scripts/new-example.ts

```diff
@@ -0,0 +1,271 @@
+        alias: 't',
+        describe: 'choose either simple type example or complex requiring module',
+        choices: exampleTypes,
+        required: false
+    })
+    .option('requiredPackages', {
+        alias: 'r',
+        describe: 'are there any additional NPM packages (other than what Cashmere already depends on) required by this example?',
+        type: 'array',
+        required: false
+    })
+    .help('help').argv;
+
+async function promptForMissingArguments() {
+    const input = await inquirer.prompt([
+        {
+            name: 'category',
+            message: `Is this example for a component in the main Cashmere library or a Cashmere Bit?`,
+            type: 'list',
+            choices: categories,
+            when: () => !args.category
+        },
+        {
+            name: 'component',
+            message: `Which component is this example for?`,
+            type: 'list',
+            choices: x => (x.category === 'bit' ? bitComponents : cashmereComponents),
+            // when the component isn't specified in args or the component that is specified in args is invalid for the specified category
+            when: x =>
+                !args.component || !((x.category || args.category) === 'bit' ? bitComponents : cashmereComponents).includes(args.component)
+        },
+        {
+            name: 'name',
+            message: 'What is the name of this example?',
+            type: 'input',
+            validate: x => (!!paramCase(x) && !existingExamples.includes(paramCase(x))) || 'Example name required and must be unique.',
+            when: () => !args.name || existingExamples.includes(paramCase(args.name))
+        },
+        {
+            name: 'type',
+            message:
+                'Is this a simple example (self-contained within single component) or do you need an entire module file for this example?',
+            type: 'list',
+            choices: exampleTypes,
+            when: () => !args.type
+        },
+        {
+            name: 'requiredPackages',
+            message: `Specify any new NPM packages (other than the following) required by this example as a comma-separated list:`,
+            suffix: currentExampleDependencies.join(','),
+            type: 'input',
+            required: false,
+            // Only prompt if other prompts have happened.  Don't want to prompt if all other required params were given as args.
+            when: x => Object.keys(x).length,
+            // transforms the user input from comma-separated string to an array of strings
+            filter: x => (x || '').split(',')
+        }
+    ]);
+
+    args = {...args, ...input};
+    args.name = paramCase(args.name);
+    args.requiredPackages = (args.requiredPackages || [])
+        .map((p: string) => p!.trim().toLowerCase())
+        .filter(p => !currentExampleDependencies.includes(p));
+
+    prettierConfig = (await prettier.resolveConfig(__dirname))!;
+}
+
+(async function newExample() {
```

Comment:
This is the main algorithm for creating a new example:
* gather any missing information from the user
* create the new example files
* register the new example with the document items service
* install any additional NPM packages the new example needs to work
* run the build to make sure we are in a good state and that the example module is regenerated

The user should be ready to `npm start` after running this.